apps/prairielearn/src/pages/instructorCopyPublicCourseInstance/instructorCopyPublicCourseInstance.ts

```diff
@@ -12,7 +12,11 @@ const router = Router();
 router.post(
   '/',
   asyncHandler(async (req, res) => {
-    const courseInstance = await selectOptionalCourseInstanceById(req.body.course_instance_id);
+    const courseInstance = await selectOptionalCourseInstanceById({
```

Comment:
If a course instance is public, it skips authzData checks. This should make this safe to use.

Reply:
IMO the new code actually makes it harder to reason about what's happening here, and could potentially impact correctness in unexpected ways in the future. While your comment may be true, it's not at all obvious. `share_source_publicly` is the _only_ thing that is necessary to check here.